### DISCUSSION TITLE
Load testing Grant Explorer

### DISCUSSION THREAD

**Discussion by shavinac** [gitcoinco/grants-stack#585] [Ideas]
### Overview
As part of productionalizing Round Manager / Grant Explorer before running alpha rounds, we want to load test our applications so we can understand how our applications behave under heavy traffic. 
Specifically, we expect **Grant Explorer** to have the heaviest traffic / usage.

### Past Efforts
On Gitcoin Passport, we tried to set up a load test of the main Passport site.
Relevant stories: https://github.com/gitcoinco/passport/issues/294 , https://github.com/gitcoinco/passport/issues/112

However, we were unsuccessful in setting up an automated load test because we were unable to automate the in-browser wallet transactions (Passport required users to sign stamps with their wallet)

Instead, we did a "manual" load test by asking the GitcoinDAO members to visit and sign up for Passport.

Unfortunately, we ran into several production issues/outages 😔 : 
* During GR14, our hosted Ceramic node was not anchoring properly, leading to lost user stamps/passports. Also our Ceramic node was repeatedly exceeding memory limits and crashing/restarting. [Story tracking Ceramic node related issues](https://github.com/gitcoinco/passport/issues/232)
* [Fleek had a production issue](https://twitter.com/FleekHQ/status/1542163010621407234) with their internal DNS resolution, leading to users unable to use Passport (which was using a custom domain)

### Possible Approaches

**Test Individual Services**
We could load-test individual layers of our application stack and/or talk with external services to get a better idea of how they handle load:

External services:
* The Graph: talk to the team team about load handling, run an automated load test on the hosted service API 
* Fleek hosting: talk to team about load for their IPFS node and their internal DNS system. Look into plans/pricing
* Pinata IPFS service: look into plans/pricing

For the centralized QF Calculation service (that the RM/GE team will build), we could also load-test that system
* test how it handles a flood of votes -- definitely should load-test this situation to make sure we can support many simultaneous voters
* test how it handles a flood of requests for the current distribution for round(s) -- less important at the moment if Round Manager app will be the main requester for distribution info

**Manual Load Testing**
We could ask GitcionDAO or other large groups of people to visit Grant Explorer and vote

**Comment by vacekj** [comment]
We still need load testing of our new indexer/api.

**Comment by vacekj** [comment]
Closing as stale
